config/environments/production.rb

```diff
@@ -18,7 +18,14 @@
   config.lograge.logger = ActiveSupport::Logger.new \
     "#{Rails.root}/log/logstash_#{Rails.env}.json"
 
-  config.staff_url = ENV.fetch('STAFF_SERVICE_URL')
+  config.staff_url = if ENV['HEROKU_APP_NAME']
+                URI.parse("https://#{ENV['HEROKU_APP_NAME']}.herokuapp.com")
+              else
+                URI.parse(ENV.fetch('STAFF_SERVICE_URL'))
```

Comment:
the URI.parse is also redundant
better to save the `url` as string in a variable and then parse the result once with URI.parse

Reply:
staff_url = if ENV['HEROKU_APP_NAME']
  "https://#{ENV['HEROKU_APP_NAME']}.herokuapp.com"
else
  ENV.fetch('STAFF_SERVICE_URL')
end

config.staff_url = URI.parse(staff_url)
# ...
```